app/src/main/java/com/github/steroidteam/todolist/todo/Task.java

```diff
@@ -0,0 +1,64 @@
+package com.github.steroidteam.todolist.todo;
+
+import java.io.Serializable;
+
+/**
+ * A class that represents a task described by a title and a body.
+ */
+public class Task implements Serializable{
+
+    private String body;
+    private final String id;
+
+    /**
+     * Constructs a new Task with a title and a body.
+     *
+     * @param id The id of the task.
+     * @param body The body of the task.
+     * @throws IllegalArgumentException Thrown if one or more arguments are null
+     */
+    public Task(String id, String body) {
```

Comment:
I am wondering if it makes sense to keep the ID in each `Task` object, or at the very least if we can maybe provide the ID afterwards.

My concern is that the ID is usually task-agnostic: it mostly depends on the database and the elements it already contains. We probably will not know the ID assigned to this task until we try to store it in the database, and its get assigned an ID.

This is problematic, because that means that we would not be able to do things like this:

```java
// Create "database"...

Task myTask = new Task("Do something"); // No ID supplied.
database.addTask(myTask); // ID assigned here.
```

A possible fix I can think of is to allow `id` to be `null` upon creation, and then let the database assign it. Not sure if it is the optimal pattern though. Thoughts?

Reply:
I see what you mean.
I remove the Id from the constructors and so it is assigned by the database when there is a request to put the task into it.
I created an Id manager (for now very simple just an incrementation) that is responsible to assign a valid id to a task. 
So we can work on the database and the task more independently. We will see how goes the Id management with the authentication system.